pkg/broker/jobs.go

```diff
@@ -138,120 +149,134 @@ func (j *apbJob) createJobMsg(
 	return jobMsg
 }
 
-// ProvisionJob - Job to provision.
-type ProvisionJob struct {
-	serviceInstance *apb.ServiceInstance
+type workFactory struct {
+}
+
+// NewWorkFactory will return a work factory capable of creating different kinds of work
+func NewWorkFactory() WorkFactory {
+	return &workFactory{}
 }
 
-// Run - Run the provision job.
-func (j *ProvisionJob) Run(token string, msgBuffer chan<- JobMsg) {
-	job := apbJob{
-		executor:               apb.NewExecutor(),
-		serviceInstanceID:      j.serviceInstance.ID.String(),
-		specID:                 j.serviceInstance.Spec.ID,
-		method:                 apb.JobMethodProvision,
-		metricsJobStartHook:    metrics.ProvisionJobStarted,
-		metricsJobFinishedHook: metrics.ProvisionJobFinished,
-		skipExecution:          false,
-		run: func(exec apb.Executor) <-chan apb.StatusMessage {
-			return exec.Provision(j.serviceInstance)
+// ProvisionJob will setup a Work implementation that will perform the provision work
+func (wf *workFactory) ProvisionJob(si *apb.ServiceInstance) Work {
```

Comment:
I'm not a fan of using the old struct name as a function now. It feels confusing to see something that looks like a "struct" vs something that is a method. For example when I see `ProvisionJob` I'd expect to see `{}` after it like: `ProvisionJob{id: "foo", blah blah blah}`.

If we're going to keep these I prefer making them look like constructor methods: `NewProvisionJob`, `NewBindJob`, etc.

Reply:
Sure happy to change these